var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -320,38 +320,42 @@ def install(self, spec, prefix):
 
     @run_after("install")
     def cache_test_sources(self):
+        if "+mpi" not in self.spec:
+            print("HYPRE must be installed with +mpi to cache test sources")
+            return
+
+        # Customize the examples makefile before caching it
+        makefile = join_path(self.extra_install_tests, "Makefile")
+        filter_file(r"^HYPRE_DIR\s* =.*", f"HYPRE_DIR = {self.prefix}", makefile)
+        filter_file(r"^CC\s*=.*", f"CC = {os.environ['CC']}", makefile)
+        filter_file(r"^F77\s*=.*", f"F77 = {os.environ['F77']}", makefile)
+        filter_file(r"^CXX\s*=.*", f"CXX = {os.environ['CXX']}", makefile)
```

Comment:
I wasn't sure if this was something you might want to do here or in `hypre-cmake` so I left it here for discussion purposes.

Reply:
Yes, we need this here. Thanks!